sdk/cosmos/azure-cosmos-spark_3-1_2-12/docs/migration.md

```diff
@@ -0,0 +1,64 @@
+| query_disableruperminuteusage | n/a | Not relevant anymore - just remove this config entry |
+| query_emitverbosetraces | n/a | Not relevant anymore - just remove this config entry |
+| query_maxbuffereditemcount | n/a | Not relevant anymore - just remove this config entry |
+| consistencylevel | spark.cosmos.read.forceEventualConsistency | By default the new connector will use Eventual Consistency for all read operations. If instead you want read operations to use the default account's consistency-level you can override the `spark.cosmos.read.forceEventualConsistency` property to `false`. |
+| rollingchangefeed | n/a | Not relevant anymore - just remove this config entry |
+| changefeedusenexttoken | n/a | Not relevant anymore - just remove this config entry |
+| changefeedusenexttoken | n/a | Not relevant anymore - just remove this config entry |
+| writestreamretrypolicy.* | n/a | Not relevant anymore - or not supported yet (poison message handling) |
+| resourcetoken | | Not supported yet with the new connector |
+| connectionmaxpoolsize | n/a | Not relevant anymore - just remove this config entry |
+| connectionidletimeout | n/a | Not relevant anymore - just remove this config entry |
+| connectionrequesttimeout | n/a | Not relevant anymore - just remove this config entry |
+
+## Conceptual differences
+The list above shows the new names of the different config options. Below you can find an explanation for conceptual differences in the new connector. The Cosmos DB Connector for Spark 3 is implementing using the DataSource V2 API (vs. Data Source V1 with the old connector) which is driving some changes especially for structured streaming.
+
+### Partitioning
+When reading data in Spark via a DataSource, the number of partitions in the returned RDD is determined  by the DataSource. In the Spark 2.4 version of the Cosmos DB connector the DataSource would create 1 Spark partition for each physical Cosmos DB partition. Especially for smaller Cosmos accounts this often lead to situations where the number of resulting Spark partitions was lower than the number of Cores available for the executors - so the latency of the spark job was relatively high because not all Executors would participate in processing teh read operations from Cosmos. The main reason for this decision was that it wasn't possible to scope a query to just a fragment of a physical partition when the Cosmos DB Connector for Spark 2.4 was created. With the new connector for Spark 3 we now by default use a new capability available in the Cosmos DB Backend, and the Java V4 SDK that allows scoping queries to just a fragment of a physical partition. As a result the default partitioning strategy will ensure that at least as many Spark partitions as Executor Cores are created. The number of Spark partition created will depend on the number of available Cores for executors, and the storage size in each physical partition.
```

Comment:
nit: Extra spaces `etermined  by the DataSo`

Reply:
Fixed